src/scripts/main.js

```diff
@@ -1 +1,69 @@
 'use strict';
+
+document.oncontextmenu = () => false;
+
+const buttonState = { left: false, right: false };
+let resetTimeout;
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.buttons === 1) {
+      return resolve('First');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
```

Comment:
The rejection message for the first promise is hardcoded as 'First promise was rejected'. Ensure this message is appropriate for your application's context.